src/odata-to-abstract-sql.ts

```diff
@@ -1717,14 +1708,11 @@ export class OData2AbstractSQL {
 					(isAliasNode(from) && from[2] === navigation.resource.tableAlias),
 			)
 		) {
+			// Add a FROM for the extraResource only if there isn't already one
 			query.fromResource(this, navigation.resource);
 			query.where.push(navigation.where);
-			return navigation.resource;
-		} else {
-			throw new SyntaxError(
```

Comment:
Instead of throwing & try-catching whenever we try to add a duplicate FROM, we now just detect it and skip adding it.

Reply:
This changes the behavior of the lambda handling - do we know if that's safe? eg I think it changes `/x?$filter=y/any(...) or y/any(...)` and not sure if that'd be fixing or breaking it

Reply:
I added extra tests based on this comment and I didn't observe any change in the behavior.
https://github.com/balena-io-modules/odata-to-abstract-sql/pull/171/files#diff-88ff1777dda6c4e24ac1a028500d389115ff29376ff74ce83703aff34e7b4cecR1224

The one that also preserves the same behavior but I'm not sure whether that's the correct one was `/pilot?$filter=can_fly__plane/plane/${methodName}(d:d/name eq 'Concorde') or can_fly__plane/plane/any(d:d/name eq '747')`

Reply:
The way I though about it not being changing the behavior was that `AddNavigation` gets called in 3 places:
* from `AddExtraFroms`, which no longer has the `catch`, which should preserve the behavior, since by the time that we throw
* from `Lambda` & `Property` which call it with a fresh `const query = new Query();`